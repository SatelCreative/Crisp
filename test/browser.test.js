const fs = require('fs');
const path = require('path');
const webdriver = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');

const loggingPrefs = new webdriver.logging.Preferences();
loggingPrefs.setLevel(
  webdriver.logging.Type.BROWSER,
  webdriver.logging.Level.ALL
);

const chromeOptions = new chrome.Options();
chromeOptions.addArguments('--start-maximized');
chromeOptions.addArguments('--headless');
chromeOptions.setLoggingPrefs(loggingPrefs);

const crisp = fs.readFileSync(
  path.resolve(__dirname, '../dist/crisp.umd.production.js'),
  'utf8'
);

let driver;

beforeAll(async () => {
  driver = new webdriver.Builder()
    .forBrowser('chrome')
    .setChromeOptions(chromeOptions)
    .build();

  await driver.get('https://satel-crisp.myshopify.com');

  await driver.executeScript(`${crisp} window.Crisp = Crisp`);
});

afterAll(() => {
  driver.quit();
});

jest.setTimeout(30000);

describe('Browser Testing', () => {
  const fileNames = fs.readdirSync(path.resolve(__dirname, 'browser'));

  for (name of fileNames) {
    if (!name.endsWith('.js')) {
      continue;
    }

    const filePath = path.resolve(__dirname, 'browser', name);
    const formattedName = name.replace('.js', '');

    it(`browser/${formattedName} generates expected output`, async () => {
      const response = await driver.executeAsyncScript(
        fs.readFileSync(filePath, 'utf8')
      );

      // Capture the logs
      const logs = await driver
        .manage()
        .logs()
        .get(webdriver.logging.Type.BROWSER);

      if (logs.length) {
        console.log(`spec "${formattedName}" logged to the console:`);
        logs.forEach(({ level, message }) => {
          switch (level.name_) {
            case 'INFO':
              console.info(message);
              return;
            case 'WARN':
              console.warn(message);
              return;
            case 'SEVERE':
              console.error(message);
              return;
            default:
              console.log(message);
          }
        });
      }

      // Special case for version
      if (formattedName === 'version') {
        expect(typeof response).toEqual('string');
      } else {
        expect(response).toMatchSnapshot(formattedName);
      }
    });
  }
});
