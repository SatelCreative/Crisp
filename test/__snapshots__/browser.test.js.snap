// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser Testing browser/promiseGet generates expected output: promiseGet 1`] = `
Array [
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Sky Busters Eurofighter Typhoon Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Sky Busters Boeing AH-64 Apache Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox 9 Car Gift Pack",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Elite Rescue Wave Cruiser",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Treasure Truck",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Grow Pro Truck",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Cookstruction Company Mega Mixer Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Garage 9-Car Pack Gift Set",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Jurassic World Dino Transporters Tyranno-Hauler Vehicle And Figure",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Jurassic World Dino Transporters T. Rex Trailer",
  },
]
`;

exports[`Browser Testing browser/requestCancelled generates expected output: requestCancelled 1`] = `"Cancelled Successfully!"`;

exports[`Browser Testing browser/requestSimple generates expected output: requestSimple 1`] = `
Object {
  "page": 1,
  "pages": 21,
  "payload": Array [
    Object {
      "tags": Array [
        "digimon",
        "egnition-sample-data",
        "stuffed-toys",
        "toys",
      ],
      "title": "10inch Anime Cute Digital Monster Digimon Koromon Soft Plush Toy Doll Pillow",
    },
    Object {
      "tags": Array [
        "american-girl",
        "egnition-sample-data",
        "on-sale",
        "toys",
      ],
      "title": "2014 Holiday Barbie Ornament African-American",
    },
    Object {
      "tags": Array [
        "digimon",
        "egnition-sample-data",
        "stuffed-toys",
        "toys",
      ],
      "title": "4\\"10cm Digimon Tailmon Plush Toy",
    },
    Object {
      "tags": Array [
        "egnition-sample-data",
        "tomy-toys",
        "toys",
      ],
      "title": "Activity Auto",
    },
    Object {
      "tags": Array [
        "egnition-sample-data",
        "toys",
        "toys-r-us",
      ],
      "title": "Animal Zone Foam T-Rex",
    },
    Object {
      "tags": Array [
        "egnition-sample-data",
        "toys",
        "toys-r-us",
      ],
      "title": "Animal Zone Stegosaurus",
    },
    Object {
      "tags": Array [
        "digimon",
        "egnition-sample-data",
        "stuffed-toys",
        "toys",
      ],
      "title": "Anime Cartoon Digimon Digital Monsters",
    },
    Object {
      "tags": Array [
        "digimon",
        "egnition-sample-data",
        "stuffed-toys",
        "toys",
      ],
      "title": "Anime Digimon Adventure Patamon Stuffed Toy Plush Toys",
    },
    Object {
      "tags": Array [
        "digimon",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Anime Digimons Digital Monster Tanemon Plush Toys",
    },
    Object {
      "tags": Array [
        "egnition-sample-data",
        "hatchimals",
        "toys",
      ],
      "title": "Authentic Hatchimals CollEGGtibles 12-Pack Egg Carton",
    },
    Object {
      "tags": Array [
        "dragon-ball",
        "egnition-sample-data",
        "toys",
      ],
      "title": "BANPRESTO BWFC DRAGON BALL Z V6 Trunks",
    },
    Object {
      "tags": Array [
        "dragon-ball",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Banpresto Dxf Dragon Ball Z 5.9-Inch Raditz Action Figure",
    },
    Object {
      "tags": Array [
        "dragon-ball",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Banpresto Resolution of Soldiers Dragon Ball Z Mr. Satan Hercule",
    },
    Object {
      "tags": Array [
        "barbie",
        "doll-accessories",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie 2-Story House",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie and Ken Dolls",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Art Studio Playset",
    },
    Object {
      "tags": Array [
        "barbie",
        "doll-accessories",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Bakery Chef Doll and Playset",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Bistro Cart",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Color Surprise Doll",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Crayola Color-In Fashion Doll and Fashions",
    },
    Object {
      "tags": Array [
        "barbie",
        "doll-accessories",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Crayola Rainbow Design Fashion Set",
    },
    Object {
      "tags": Array [
        "barbie",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Crayola Tie-Dye Fashions",
    },
    Object {
      "tags": Array [
        "barbie",
        "doll-accessories",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie DreamCamper",
    },
    Object {
      "tags": Array [
        "barbie",
        "doll-accessories",
        "dolls",
        "egnition-sample-data",
        "toys",
      ],
      "title": "Barbie Dreamhouse",
    },
  ],
  "size": 24,
  "total": 500,
}
`;

exports[`Browser Testing browser/simpleGet generates expected output: simpleGet 1`] = `
Array [
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Sky Busters Eurofighter Typhoon Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Sky Busters Boeing AH-64 Apache Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox 9 Car Gift Pack",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Elite Rescue Wave Cruiser",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Treasure Truck",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Grow Pro Truck",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Cookstruction Company Mega Mixer Vehicle",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "on-sale",
      "toys",
    ],
    "title": "Matchbox Garage 9-Car Pack Gift Set",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Jurassic World Dino Transporters Tyranno-Hauler Vehicle And Figure",
  },
  Object {
    "tags": Array [
      "egnition-sample-data",
      "matchbox",
      "toys",
    ],
    "title": "Matchbox Jurassic World Dino Transporters T. Rex Trailer",
  },
]
`;
